src/features/offer/api/useSimilarOffers.ts

```diff
@@ -1,35 +1,162 @@
+) => {
+  const relatedProducts = await getRelatedProducts({
+    recommendClient,
+    indexName,
+    objectIDs: [String(offerId)],
+    queryParameters,
+    fallbackParameters,
+  })
+    .then((response) => response.recommendations)
+    .then((recommendations) => recommendations.map((recommendation) => recommendation.objectID))
+    .catch((error) => eventMonitoring.captureException(error))
+
+  return typeof relatedProducts === 'string' ? [relatedProducts] : relatedProducts || []
+}
+
+const getAlgoliaFrequentlyBoughtTogether = async (
+  offerId: string,
+  queryParameters: RecommendSearchOptions
+) => {
+  const frequentlyBoughtTogether = await getFrequentlyBoughtTogether({
+    recommendClient,
+    indexName,
+    objectIDs: [String(offerId)],
+    queryParameters,
+  })
+    .then((response) => response.recommendations)
+    .then((recommendations) => recommendations.map((recommendation) => recommendation.objectID))
+    .catch((error) => eventMonitoring.captureException(error))
+
+  return typeof frequentlyBoughtTogether === 'string'
+    ? [frequentlyBoughtTogether]
+    : frequentlyBoughtTogether || []
+}
+
+const getApiRecoSimilarOffers = async (similarOffersEndpoint: string) => {
+  const similarOffers: string[] = await fetch(similarOffersEndpoint)
+    .then((response) => response.json())
+    .then((data) => data.results)
+    .catch((error) => eventMonitoring.captureException(error))
+
+  return similarOffers
+}
+
+export const useSimilarOffers = ({
+  offerId,
+  position,
+  shouldUseAlgoliaRecommend,
+  categoryIncluded,
+  categoryExcluded,
+}: Props) => {
+  const { data } = useSubcategories()
+
+  const categories: SearchGroupNameEnumv2[] = useMemo(() => {
+    if (categoryIncluded) {
+      return [categoryIncluded]
+    }
+
+    return (
+      data?.searchGroups
+        .filter(
+          (searchGroup) =>
+            searchGroup.name !== categoryExcluded && searchGroup.name !== SearchGroupNameEnumv2.NONE
+        )
+        .map((searchGroup) => searchGroup.name) || []
+    )
+  }, [categoryExcluded, categoryIncluded, data?.searchGroups])
+
   const { user: profile } = useAuthContext()
-  const similarOffersEndpoint = getSimilarOffersEndpoint(offerId, profile?.id, position) as string
+  const similarOffersEndpoint = getSimilarOffersEndpoint(
+    offerId,
+    profile?.id,
+    position,
+    categories
+  ) as string
   const [similarOffersIds, setSimilarOffersIds] = useState<string[]>()
 
-  useEffect(() => {
-    if (!similarOffersEndpoint) return
+  const fetchAlgolia = useCallback(async () => {
+    const { queryParameters, fallbackParameters } = getAlgoliaRecommendParams(position, categories)
+    if (categoryIncluded) {
+      setSimilarOffersIds(
+        await getAlgoliaRelatedProducts(String(offerId), queryParameters, fallbackParameters)
+      )
+    } else {
+      setSimilarOffersIds(
+        await getAlgoliaFrequentlyBoughtTogether(String(offerId), queryParameters)
+      )
+    }
```

Comment:
```suggestion
    setSimilarOffers(categoryIncluded
      ? await getAlgoliaRelatedProducts(String(offerId), queryParameters, fallbackParameters)
      : await getAlgoliaFrequentlyBoughtTogether(String(offerId), queryParameters)
    )
```

Je ne sais pas si ça perd en lisibilité 🤔 
Mais j'aime bien cette façon d'écrire

Reply:
Franchement ça passe bien vu je vais update 👍